### ISSUE TITLE
bzz://theswarm.eth doesn't work

### ISSUE THREAD

**Issue by oskarth** [status-im/status-react#8104]
# Problem
bzz://theswarm.eth doesn't resolve

# Expected behavior
It resolves, probably using Swarm gateway

OR link to issue on "We don't yet have Swarm support, we'd love to, here's the <GHI> you can follow along"

## Notes
Customer feedback https://www.reddit.com/r/ethereum/comments/bjhptq/status_release_0120_introducing_discover_v1_and/

cc @mandrigin @flexsurfer @yenda @annadanchenko fyi

**Comment by flexsurfer** [comment]
We resolve ens according to EIP 1577 https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1577.md, and `theswarm.eth` doesn't  follow it https://manager.ens.domains/name/theswarm.eth, and also we have a small bug in resolution code which will be addressed here https://github.com/status-im/status-react/pull/8042

**Comment by annadanchenko** [comment]
not our bug

**Comment by oskarth** [comment]
If the problem is upstream, can we please be good Ethereum and OSS citizens and open a bug upstream?

**Comment by oskarth** [comment]
@mandrigin Don't know how you label upstream bugs, but once we have a corresponding ticket I guess we want to indicate that this is blocked by upstream bug / wontfix etc.

**Comment by oskarth** [comment]
I was trying to find upstream but don't know where it is, could open in go-ethereum I guess but probably better place.

@decanus wdyt?

**Comment by mandrigin** [comment]
@oskarth I'm a bit out of context here, but why is at an upstream bug? I thought @flexsurfer meant that the domain isn't set up according to standards.

**Comment by oskarth** [comment]
> domain isn't set up according to standard

The standard is new, and standards are changing. Swarm has existed for longer than that standard. They probably want to support that standard, but someone needs to tell them. This is not just about them but about our users. Swarm is a big Ethereum project and people bump into this in real life and think this is the fault of Status.

If they can't / won't, and we definitely can't resolve it for some reason (?), perhaps we can show a basic warning, like we only support EIPXX resolution, or whatever.

**Comment by mandrigin** [comment]
so that sounds like 2 issues: one is to show a warning that is a part of our UI and another is to ping them?
where do you think is the best place to ask these questions to the Swarm guys and gals?

**Comment by decanus** [comment]
@oskarth I think its best to find out who owns this name and then update it, I guess we could easily ping the swarm guys.

Depending on what resolver they use, the can still support EIP-1577 and their old resolution

**Comment by oskarth** [comment]
Asked here https://gitter.im/ethersphere/orange-lounge?at=5cd9052bbdc3b64fcf39331d

**Comment by acud** [comment]
The ENS entry is indeed in the old resolver format. The EIP specified that a transition period should exist while maintaining backwards compatibility and we assume to still be in that transition period (the period length is not mentioned in the EIP). Swarm supports ENS resolution through both the updated EIP resolver _and_ the old resolver.

**Comment by oskarth** [comment]
Thanks @justelad! Do you have any idea roughly of how long you currently assume the transition period to be, given that this is currently underspecified? This might inform how much users and dapp browers should rely on this standard being implemented.

@flexsurfer is there any technical reason we wouldn't support the old resolver, given the information provided above?

**Comment by acud** [comment]
@oskarth I arbitrarily shoot to discontinue the old resolver support later this year, somewhere around July probably. Supporting the old contract right now comes at zero cost from our side so we're also not so pressed to deprecate it.

**Comment by flexsurfer** [comment]
@oskarth  i'm not sure if there is a standard for the `content` field so we could support it, we supported it for ipfs because there were a few Dapps on ipfs which used ens names and we still maintaining backwards compatibility for the `content` field for ipfs, but EIP says ` This condition MUST be enforced until at least March 31st, 2019.`

**Comment by cobordism** [comment]
I have migrated theswarm.eth to the new registrar and also updated to the new resolver format.
It should work now for you.

**Comment by flexsurfer** [comment]
great, thanks @homotopycolimit !

**Comment by flexsurfer** [comment]
we need to add swarm support here https://github.com/status-im/status-react/blob/b0449f3416aa0a78b88a23bc67a6e3f2b65e090f/src/status_im/utils/contenthash.cljs#L26

**Comment by acud** [comment]
also to update you all, the support for the new ENS resolver is now live as part of the `0.4` Swarm release. so you can use your Swarm nodes to resolve the ENS entries that use the new public resolver

**Comment by flexsurfer** [comment]
great, thanks @justelad !
